core/src/main/java/org/aerogear/mobile/core/http/pinning/CertificatePinningCheck.java

```diff
@@ -0,0 +1,85 @@
+package org.aerogear.mobile.core.http.pinning;
+
+import android.support.annotation.NonNull;
+
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+
+/**
+ * Class for performing certificate pinning checks.
+ *
+ * This consists of a HTTP GET request using a provided {@link HttpServiceModule}. If the request is
+ * successful then a success listener will be invoked. If unsuccessful then a failure listener will
+ * be invoked.
+ *
+ * If a listener is attached after the request is already completed then the listener will be
+ * invoked immediately with the result of the request being provided.
+ */
+public class CertificatePinningCheck {
```

Comment:
@secondsun This PR is still early on, based on another PR which will be closed, should be fleshed out properly tomorrow. Would you mind taking a look at this class, is this what you meant when we spoke about the listener pattern?